apps/bluetooth/test/unit/modules/transfer_manager_test.js

```diff
@@ -0,0 +1,678 @@
+          mockShowConfirmPromise = Promise.resolve(mockUserAction);
+          this.sinon.stub(TurnBluetoothOnDialog, 'showConfirm').returns(
+            mockShowConfirmPromise);
+        });
+
+        test('TurnBluetoothOnDialog should be shown with confirm dialog, ' +
+             '_endTransferWithReason should be called with reason "cancelled" ',
+             function(done) {
+          TransferManager._onBluetoothEnabledChanged(mockEnabledState);
+          assert.isFalse(TurnBluetoothOnDialog.close.called);
+          assert.isTrue(TurnBluetoothOnDialog.showConfirm.called);
+          mockShowConfirmPromise.then((userAction) => {
+            assert.equal(userAction, mockUserAction);
+            assert.isFalse(BtContext.setEnabled.called);
+            assert.isTrue(
+              TransferManager._endTransferWithReason.calledWith('cancelled'));
+          }, () => {
+            // reject case
+          }).then(done, done);
+        });
+      });
+    });
+  });
+
+  suite('_watchEventFromDevicePicker > ', function() {
+    setup(function() {
+      this.sinon.stub(DevicePickerPanel, 'addEventListener');
+    });
+
+    test('Should add "devicePicked", "cancelSelection" events, ' +
+         'and regited _onDevicePicked(), _onCancelSelection() handlers ' +
+         'from DevicePickerPanel ', function() {
+      TransferManager._watchEventFromDevicePicker();
+      assert.equal(DevicePickerPanel.addEventListener.args[0][0],
+        'devicePicked');
+      assert.isDefined(DevicePickerPanel.addEventListener.args[0][1]);
+      assert.equal(DevicePickerPanel.addEventListener.args[1][0],
+        'cancelSelection');
+      assert.isDefined(DevicePickerPanel.addEventListener.args[1][1]);
+    });
+  });
+
+  suite('_onDevicePicked > ', function() {
+    var mockActivityRequest, mockEvent, mockSchedule;
+    setup(function() {
+      mockSchedule = {};
+      this.sinon.stub(TransferManager,
+                      '_produceSendingFilesSchedule').returns(mockSchedule);
+      this.sinon.stub(TransferManager, '_postMessageToSystemApp');
+      this.sinon.stub(BtContext, 'sendFile');
+      mockEvent = {
+        detail: {
+          address: 'AA:BB:CC:00:11:22'
+        }
+      };
+    });
+
+    suite('blob has name > ', function() {
+      setup(function() {
+        mockActivityRequest = {
+          source: {
+            name: 'share',
+            data: {
+              blobs: [{
+                name: 'file_01'
+              }, {
+                name: 'file_02'
+              }]
+            }
+          }
+        };
+        TransferManager._activity = mockActivityRequest;
+      });
+
+      test('_produceSendingFilesSchedule() should be called with length, ' +
+           '_postMessageToSystemApp() should be called with schedule, ' +
+           'BtContext.sendFile() should be called with: ' +
+           'address of target device and blob in soure.data ', function() {
+        TransferManager._onDevicePicked(mockEvent);
```

Comment:
In addition to ensure all of the blobs are sent, we also need to make sure `_endTransferWithReason` is called correctly. You may want to return the promise from `_onDevicePicked` so that you have of correct timing doing the assertions.